src/core/queryCache.ts

```diff
@@ -20,7 +20,18 @@ interface QueryHashMap {
   [hash: string]: Query<any, any>
 }
 
-type QueryCacheListener = (query?: Query) => void
+interface QueryCacheNotifyMeta {
```

Comment:
If you define the event separately you can specify which additional properties it has, like:
```ts
interface QueryAddedEvent {
  type: 'queryAdded'
  query: Query
}

interface QueryUpdatedEvent {
  type: 'queryUpdated'
  query: Query
  action: Action
}

type Events = QueryAddedEvent | QueryUpdatedEvent
```
This way TS can check if the correct properties are set when defining an event, but it can also be used when using an event:
```ts
if (event.type === 'queryUpdated') {
  console.log(event.action) // TS knows action is there
}
```

Reply:
Thanks for the tip. I'll implement this right now :)